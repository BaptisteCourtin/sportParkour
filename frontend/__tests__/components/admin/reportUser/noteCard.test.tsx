import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

import { ReportStatus } from "@/types/graphql";

import NoteCard from "@/components/admin/reportUser/noteCard";

jest.mock("next/router", () => require("next-router-mock"));

jest.mock(
  "@/components/admin/buttonForComment",
  () =>
    ({ malfratId, parkourId, commentaire, isAdmin, isClient }: any) =>
      <button data-testid="button-for-comment">Button for Comment</button>
);

describe("NoteCard", () => {
  const mockNote = {
    id: "1",
    status: ReportStatus.NonVu,
    commentaire: "Commentaire en faute",
    parkour: {
      id: "parkour-1",
      title: "Parkour Title",
    },
  };

  const mockMalfratId = "111";

  it("matches snapshot", () => {
    const { asFragment } = render(
      <NoteCard note={mockNote} malfratId={mockMalfratId} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly", () => {
    render(<NoteCard note={mockNote} malfratId={mockMalfratId} />);

    // Vérifier la présence du commentaire en faute
    expect(screen.getByText("Commentaire en faute")).toBeInTheDocument();

    // Vérifier la présence du bouton pour commentaire
    expect(screen.getByTestId("button-for-comment")).toBeInTheDocument();

    // Vérifier la présence du lien vers le parkour
    expect(
      screen.getByText("Parkour du commentaire : Parkour Title")
    ).toBeInTheDocument();
  });
});
