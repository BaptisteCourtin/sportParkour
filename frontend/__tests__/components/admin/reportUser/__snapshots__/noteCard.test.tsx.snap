// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteCard matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="NON_VU noteCard"
  >
    <div
      class="top"
    >
      <p
        class="commentInFaute"
      >
        Commentaire en faute
      </p>
      <button
        data-testid="button-for-comment"
      >
        Button for Comment
      </button>
    </div>
    <a
      href="/parkour/parkour-1"
    >
      parkour du commentaire : Parkour Title
    </a>
  </li>
</DocumentFragment>
`;
